### ISSUE TITLE
FLY002 wants to make my code less readable

### ISSUE THREAD

**Issue by mgedmin** [astral-sh/ruff#27635]
### Summary

I have a unit test with an assertion like this:
```python
    assert output.getvalue().rstrip('\n') == '\n'.join([
        '   verb   noun   wh',
        '>  hello  world  no',
        '-  bye    world  ha'
    ])
```

ruff thinks I should be using an f-string instead of string joins because apparently

```python
    assert output.getvalue().rstrip('\n') == '  verb  noun  wh\n> hello world no\n- bye   world ha'
```

is somehow "more readable".

```console
$ ruff check --isolated --select FLY example.py
FLY002 Consider f-string instead of string join
 --> example.py:1:42
  |
1 |   assert output.getvalue().rstrip('\n') == '\n'.join([
  |  __________________________________________^
2 | |     '   verb   noun   wh',
3 | |     '>  hello  world  no',
4 | |     '-  bye    world  ha'
5 | | ])
  | |__^
help: Replace with f-string

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

(the "apparently more readable" example was produced by `ruff check --isolated --select FLY example.py --diff --unsafe-fixes`)

Since the intent of the FLY002 rule is to use f-strings for interpolation, maybe the linter should not emit this warning when all of the things being joined are string literals?

### Version

ruff 0.16.1

**Comment by ntBre** [comment]
It looks like this transformation is intentionally in scope for the rule based on https://github.com/astral-sh/ruff/issues/2102. I think I'd recommend a `noqa` comment in this case if you prefer the input code.

flynt itself also applies this change with a long enough line-length limit, so this is consistent with upstream:

```console
$ uvx flynt -tj -ll 120 - <<EOF
def foo():
    assert output.getvalue().rstrip('\n') == '\n'.join([
        '   verb   noun   wh',
        '>  hello  world  no',
        '-  bye    world  ha'
    ])
EOF
def foo():
    assert output.getvalue().rstrip('\n') == "   verb   noun   wh\n>  hello  world  no\n-  bye    world  ha"
```

**Comment by mgedmin** [comment]
It just feels super weird to have a rule explicitly bring up f-strings where the recommended fix ends up not using one, but if you want to be bug-compatible with another project, I won't gainsay that.

**Comment by Avasam** [comment]
For the sake of exploring alternatives, this is just as concise, as readable (imo more), doesn't rely on implicit concatenation and doesn't trigger any rule:

```py
import textwrap
assert output.getvalue() == textwrap.dedent("""\
       verb   noun   wh
    >  hello  world  no
    -  bye    world  ha
""")
```
or even
```py
assert output.getvalue() == """\
   verb   noun   wh
>  hello  world  no
-  bye    world  ha
"""
```

---

> have a rule explicitly bring up f-strings where the recommended fix ends up not using one

Agreed either way.

**Comment by mgedmin** [comment]
Whoa, textwrap.dedent actually works!  I was under the mistaken impression I couldn't use it because the first line had some extra spaces at the front.  I thought textwrap.dedent() would get confused, hence this ''.join() contraption.  I'll have to review my tests to see if I have any where _all_ of the lines start with spaces.
